src/shared.js

```diff
@@ -1,12 +1,34 @@
+import axios from 'axios'
+import { ref } from 'vue'
+
 export const MIN_YEAR = 1850
-export const MAX_YEAR = 2024
+export const MAX_YEAR = ref(2022)
+export const MAX_MONTH = ref(11)
+
+export let xrange = []
+
+export async function getMaxYearAndMonth() {
```

Comment:
I think this logic all belongs in the Store now because it's doing an async request which is exactly what the shared state / action in Stores is for.